hafnian/tests/test_samples.py

```diff
@@ -201,26 +201,30 @@ def test_probability_vacuum(self):
 
         assert np.abs(prob0 - prob0_estimate) < delta
 
-    def test_multimode_vacuum_state_hafnian(self):
+    @pytest.mark.parametrize("sample_func", [hafnian_sample_state, hafnian_sample_classical_state])
+    def test_multimode_vacuum_state_hafnian(self, sample_func):
         """Test the sampling routines by checking the samples for pure vacuum
         using the sampler for classical states
         """
         n_samples = 100
         n_modes = 10
         sigma = np.identity(2 * n_modes)
         zeros = np.zeros(n_modes, dtype=np.int)
-        samples = hafnian_sample_classical_state(sigma, samples=n_samples)
+        samples = sample_func(
+            sigma, samples=n_samples
+        )  # hafnian_sample_classical_state(sigma, samples=n_samples)
         for i in range(n_samples):
             assert np.all(samples[i] == zeros)
 
-    def test_multimode_vacuum_state_hafnian_classical(self):
+    @pytest.mark.parametrize("sample_func", [hafnian_sample_state, hafnian_sample_classical_state])
+    def test_thermal_state_hafnian(self, sample_func):
         """Test the sampling routines by checking the samples for a single mode
         thermal state
         """
         n_samples = 10000
-        mean_n = 1.0
+        mean_n = 0.5
```

Comment:
How come you had to reduce this down from 1?

Reply:
For the hafnian based method for sampling one needs to also coarse grain the last measurement since one only samples up to a certain cutoff.